### PR TITLE
[Issue-33]: Slider

### LINKED ISSUE #33: Slider
- Move over slider from https://github.com/hlxsites/caesars/tree/main/caesars-palace/blocks/slider
- Simplify to use less media queries (use classes instead)
- Try to simplify build ellipsis (remove block config) to use good default values

### PR DESCRIPTION
**Pull request by swetabar**
Fix #33 

## 🔗 Test URLs:

- Before: https://main--caesars-careers--hlxsites.hlx.page/drafts/swbarman/slider
- After: https://issue-33-slider--caesars-careers--hlxsites.hlx.page/drafts/swbarman/slider

## 📝 Description:

Screen Shot 2023-05-09 at 11 20 22 AM
Screen Shot 2023-05-09 at 11 20 41 AM
Screen Shot 2023-05-09 at 11 20 58 AM

### CODE REVIEW

File: careers/blocks/slider/slider.css [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+  text-transform: uppercase;
+  line-height: 1.1;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+  overflow: hidden;
+}
+
+.slider .card .short-description h5 {
+  opacity: .7;
+  font-size: 12px;
```
Comment (tmathern) [inline careers/blocks/slider/slider.css:119]: Variable?

File: careers/blocks/slider/slider.css [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+.slider .tall-card .active-close-button {
+  position: absolute;
+  top: var(--layout-size-xs);
+  right: var(--layout-size-xxs);
+  background: url('../../icons/close-bold.svg') no-repeat;
+  min-width: var(--layout-size-xl);
+  min-height: var(--layout-size-xl);
+  transform: scale(0.8);
+  cursor: pointer;
+  z-index: 2;
```
Comment (tmathern) [inline careers/blocks/slider/slider.css:176]: Why 2?

File: careers/blocks/slider/slider.js [hunk 3]
```diff
@@ -0,0 +1,218 @@
+import { buildEllipsis } from '../../scripts/scripts.js';
```
Comment (tmathern) [inline careers/blocks/slider/slider.js:1]: @amol-anand Should we go full-CSS for this one and make the text clickable instead? So we could remove "buildEllipsis altogether".

File: careers/blocks/slider/slider.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+
+const DEFAULT_CONFIG = Object.freeze({
+  'visible-slides': 3,
+  maxlines: 3,
```
Comment (tmathern) [inline careers/blocks/slider/slider.js:5]: If we remove the `buildEllispsis`, we call also simplify this...

File: careers/blocks/slider/slider.js [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+    block.appendChild(arrowLeft);
+
+    const arrowRight = document.createElement('div');
+    arrowRight.classList.add('slider-button', 'right');
+    block.appendChild(arrowRight);
+  }
+
+  const shortDescriptionDivs = block.querySelectorAll('.short-description');
+  shortDescriptionDivs.forEach((div) => {
+    const ellipsableText = div.querySelector('p');
```
Comment (tmathern) [inline careers/blocks/slider/slider.js:57]: A lot of code here could be removed if we go full-CSS and make the text clickable.. wdyt?

**Review by tmathern — CHANGES_REQUESTED** [review]

**Review by tmathern — CHANGES_REQUESTED** [review]
How much do we want to simplify the ellipsis building?

File: careers/blocks/slider/slider.js (hunk 4 above)
Comment (swetabar) [reply careers/blocks/slider/slider.js:3]: Done.

File: careers/blocks/slider/slider.js (hunk 5 above)
Comment (swetabar) [reply careers/blocks/slider/slider.js:55]: Done!

File: careers/blocks/slider/slider.js (hunk 3 above)
Comment (swetabar) [reply careers/blocks/slider/slider.js]: Done.

File: careers/blocks/slider/slider.css (hunk 1 above)
Comment (swetabar) [reply careers/blocks/slider/slider.css]: Done.

**Review by amol-anand — APPROVED** [review]

**Pull request merged**